### ISSUE TITLE
Soil turnover times on parameter file and hard coded?

### ISSUE THREAD

**Issue by wwieder** [ESCOMP/CTSM#995]
Here it looks like the soil turnover times are being read from the parameter file in `DecompCascadeBGCreadNML`
https://github.com/ESCOMP/CTSM/blob/605ee9dc7c61219498f50d25bde29decc87bc4f2/src/soilbiogeochem/SoilBiogeochemDecompCascadeBGCMod.F90#L183

but then they are hard coded in `decomp_rate_constants_bgc`
https://github.com/ESCOMP/CTSM/blob/605ee9dc7c61219498f50d25bde29decc87bc4f2/src/soilbiogeochem/SoilBiogeochemDecompCascadeBGCMod.F90#L730

Values here look the same, but this may complicated the PPE @dlawrenncar wants to do?

**Comment by wwieder** [comment]
Similarly for nitrification`k_nitr_max` is on the parameter file, which limits nitrification rates (per second), but as far as I can tell, the code actually uses `k_nitr_max_perday`, which is a namelist variable.

https://github.com/ESCOMP/CTSM/blob/1239f17f07dbce20f6aea2cb120744dc07a1ded2/src/soilbiogeochem/SoilBiogeochemNitrifDenitrifMod.F90#L149

**Comment by billsacks** [comment]
I agree that it looks like `k_nitr_max` is never actually read from the netcdf file.

Regarding your first comment, I see this, which seems to explain the history of it:

https://github.com/ESCOMP/CTSM/blob/605ee9dc7c61219498f50d25bde29decc87bc4f2/src/soilbiogeochem/SoilBiogeochemDecompCascadeBGCMod.F90#L740-L751

**Comment by wwieder** [comment]
A few questions then as to the most efficient / appropriate way to: 

1. Remove legacy parameters from the parameter file that are not used (`k_nitr_max`)

2. Replace the hard coded parameters with namelist or parameter file values that are actually being used (e.g. `tau_s1` should be a namelist parameter = 1/18.5 (to avoid roundoff errors)).

Does this happen on @olyson's parameter branch? What about keeping  track of namelist vs. parameter files changes / duplications?

**Comment by olyson** [comment]
Regarding k_nitr_max_perday, it looks like a namelist group, nitrif_inparm, was created at some point to hold this parameter and other parameters, but it is empty.  So the model is using the hardcoded value of k_nitr_max_perday (0.1).  The other parameters in this group are also using hardcoded values: denitrif_respiration_coefficient, denitrif_respiration_exponent, denitrif_nitrateconc_coefficient, denitrif_nitrateconc_exponent.  And then these values are assigned to the params_inst% versions of these parameters.
So maybe we should populate the nitrif_inparm namelist group with these parameters and have the code use these.  That should be bfb.

**Comment by ekluzek** [comment]
We've gone back and forth on the philosophy of if we should put everything into the namelist or on the params file. I'm guessing this namelist was added when we were thinking everything should go to namelist, but wasn't completed when we decided things should go on the params file. 

At this point we actually think that things in namelist should go on the params file. At least longer term. So if you put it in namelist we will eventually want to move it to the params file. However, this is also true of other namelist parameters, so it will need to be done elsewhere. So if it's easier for you to do that now on your branch it still might be a reasonable solution.
